sqlglot/optimizer/qualify_columns.py

```diff
@@ -211,6 +211,23 @@ def _qualify_columns(scope, resolver):
             if column_table:
                 column.set("table", exp.to_identifier(column_table))
 
+    # Determine whether each reference in the order by clause is to a column or an alias.
+    order = scope.find(exp.Order)
+    if order is not None:
+        for column in order.find_all(exp.Column):
```

Comment:
Should this logic be here?: https://github.com/tobymao/sqlglot/blob/main/sqlglot/optimizer/scope.py#L235

Reply:
That is, Scope.columns has logic to exclude columns in the Order clause. Sounds like that's wrong.

This change fixes qualify_columns, but it doesn't fix everything else: Scope.columns still won't return these columns.

Reply:
I considered putting it there but we'll need access to the schema to do so. Also I assumed you intentionally left those out from scope#columns bc the reference may be to an alias, in which case qualifying it would be incorrect